### ISSUE TITLE
Replace use of deprecated github.com/golang/protobuf/proto

### ISSUE THREAD

**Issue by SpicyLemon** [provenance-io/provenance#1014]
## Summary

The `github.com/golang/protobuf/proto` package is deprecated as it has moved to `google.golang.org/protobuf/proto`.

## Problem Definition

Several files still import the deprecated package, some of them are the auto-generated `query.pb.gw.go` files.

## Proposal

* Update what's needed to have the proto go code generated to use the new package.
* Replace uses of the old package in our non-generated code.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by SpicyLemon** [comment]
I started looking into this one and I don't think there's anything we can do for now.

The imports of `github.com/golang/protobuf/proto` are in the `.pb.gw.go` files and several handler tests. The use in the handler tests can't be replaced with `google.golang.org/protobuf/proto` because they have different `proto.Message` interfaces. And since it's being generated for the deprecated library, it doesn't conform to the replacement library's stuff.

I tried replacing our `protoc` stuff used to generate code/docs with `buf generate`, but the generated code still uses the deprecated `github.com/golang/protobuf/proto` library. That work can be seen in this branch: `dwedul/1014-protobuf-depr-update`

Another hiccup I had is that the docker container we use for the protoc stuff doesn't have the `protoc-gen-doc` utility and can't generate the `docs/proto-docs.md`. So we'd need to generate/publish our own docker container with it in order to keep doing what we've got.
